ambry-store/src/main/java/com.github.ambry.store/CompactionManager.java

```diff
@@ -15,22 +15,80 @@
 
 import com.github.ambry.config.StoreConfig;
 import com.github.ambry.utils.Time;
+import com.github.ambry.utils.Utils;
+import java.util.Collection;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 
 /**
  * Responsible for managing compaction of a {@link BlobStore}. V0 implementation returns entire log segment range
  * ignoring those overlapping with {@link Journal} as part of CompactionDetails.
  */
 class CompactionManager {
+  private final String mountPath;
   private final StoreConfig storeConfig;
   private final Time time;
   private final long messageRetentionTimeInMs;
+  private final Collection<BlobStore> stores;
+  private final CompactionExecutor compactionExecutor;
+  private final Logger logger = LoggerFactory.getLogger(getClass());
 
-  CompactionManager(StoreConfig storeConfig, Time time) {
+  private Thread compactionThread;
+
+  /**
+   * Creates a CompactionManager that handles scheduling and executing compaction.
+   * @param mountPath the mount path of all the stores for which compaction may be executed.
+   * @param storeConfig the {@link StoreConfig} that contains configurationd details.
+   * @param stores the {@link Collection} of {@link BlobStore} that compaction needs to be executed for.
+   * @param time the {@link Time} instance to use.
+   */
+  CompactionManager(String mountPath, StoreConfig storeConfig, Collection<BlobStore> stores, Time time) {
+    this.mountPath = mountPath;
     this.storeConfig = storeConfig;
+    this.stores = stores;
     this.time = time;
     this.messageRetentionTimeInMs = storeConfig.storeDeletedMessageRetentionDays * Time.SecsPerDay * Time.MsPerSec;
+    compactionExecutor = storeConfig.storeEnableCompaction ? new CompactionExecutor() : null;
+  }
+
+  /**
+   * Enables the compaction manager allowing it execute compactions if required.
+   */
+  void enable() {
```

Comment:
It's not completely synonymous because we have to `disable()` and `awaitTermination()` for this. With `shutdown()`, its one call.